src/scenes/ModulesRouter/ModuleEventSessionView.tsx

```diff
@@ -0,0 +1,208 @@
+import { requestGetModule, requestGetModuleEvent } from 'src/redux/features/modules/modulesAsyncThunks';
+import { selectModule } from 'src/redux/features/modules/modulesSelectors';
+import { selectProjectModuleList } from 'src/redux/features/modules/modulesSelectors';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import strings from 'src/strings';
+import { ModuleEventType, getModuleEventName } from 'src/types/Module';
+import { getLongDateTime } from 'src/utils/dateFormatter';
+
+import ModuleViewTitle from './ModuleViewTitle';
+
+const CALL_DESCRIPTION_HTML = `
+  <div>
+    <p>Clicking "Join" will open up a browser window to join a Zoom video call.</p>
+    <p>For this Live Session you will need:</p>
+    <ul>
+      <li>An internet connection – broadband wired or wireless (3G or 4G/LTE)</li>
+      <li>Speakers and a microphone – built-in, USB plug-in, or wireless Bluetooth</li>
+      <li>A webcam or HD webcam - built-in, USB plug-in</li>
+    </ul>
+  </div>
+`;
+
+const openExternalURL = (url: string | undefined, target = '_blank', features = 'noopener noreferrer') => {
+  if (url) {
+    window.open(url, target, features);
+  }
+};
+
+const ModuleEventSessionView = () => {
+  const dispatch = useAppDispatch();
+  const { activeLocale } = useLocalization();
+  const theme = useTheme();
+  const { project, projectId } = useProject();
+  const pathParams = useParams<{ eventId: string; moduleId: string; projectId: string }>();
+  const eventId = Number(pathParams.eventId);
+  const moduleId = Number(pathParams.moduleId);
+  const module = useAppSelector(selectModule(moduleId));
+  const modules = useAppSelector(selectProjectModuleList(projectId));
+  const moduleEventType: ModuleEventType | undefined =
+    module?.events && (Object.keys(module.events)[eventId] as ModuleEventType);
+  const event = moduleEventType ? module?.events?.[moduleEventType] : undefined;
+  const eventSession = event?.sessions?.[0];
+  const eventName = moduleEventType ? getModuleEventName(moduleEventType) : '';
+
+  const [now, setNow] = useState(new Date());
+
+  const crumbs: Crumb[] = useMemo(
+    () => [
+      {
+        name: activeLocale ? 'Module' : '',
+        to: APP_PATHS.PROJECT_MODULE.replace(':projectId', `${projectId}`).replace(':moduleId', `${moduleId}`),
+      },
+    ],
+    [activeLocale, moduleId, projectId]
+  );
+
+  const eventIsStartingSoon = useMemo(() => {
+    if (eventSession?.startTime) {
+      const startTime = new Date(eventSession.startTime);
+      const diff = startTime.getTime() - now.getTime();
+      return diff < FIFTEEN_MINUTE_INTERVAL_MS;
+    }
+
+    return false;
+  }, [eventSession, now]);
+
+  const eventHasEnded = useMemo(() => {
+    if (eventSession?.endTime) {
+      const endTime = new Date(eventSession.endTime);
+      const diff = endTime.getTime() - now.getTime();
+      return diff < 0;
+    }
+
+    return false;
+  }, [eventSession, now]);
+
+  // update the current time every minute
```

Comment:
very nice 👍 
thoughts on extracting this out as a hook for reusability? useCountdown or useTimer or useClock or useDate.. maybe for later, we can get as simple or fancy here.

Reply:
🤔